From f6bdc678d55d8c75ac50025a2d1f4e716072add5 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Thu, 18 May 2023 19:58:19 +0800
Subject: [PATCH] Add some COMDAT sections

When using function/data sections option from
gcc, some additional un-packed sections get created
move these to there correct placement if not already
included

sections seen on x86/x86_64/aarch64:
.gcc_except_table.efi_main
.bss.debugging

.gcc_except_table.* only appears on objcopy archs
---
 gnuefi/elf_loongarch64_efi.lds | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnuefi/elf_loongarch64_efi.lds b/gnuefi/elf_loongarch64_efi.lds
index fccfe77..4bc0b0a 100644
--- a/gnuefi/elf_loongarch64_efi.lds
+++ b/gnuefi/elf_loongarch64_efi.lds
@@ -34,7 +34,7 @@ SECTIONS
    *(.sbss)
    *(.scommon)
    *(.dynbss)
-   *(.bss)
+   *(.bss*)
    *(COMMON)
    . = ALIGN(16);
    _bss_end = .;
